src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -548,11 +548,17 @@ void os::print_context(outputStream *st, const void *context) {
   // Add XMM registers + MXCSR. Note that C2 uses XMM to spill GPR values including pointers.
   st->cr();
   st->cr();
-  for (int i = 0; i < 16; ++i) {
-    const int64_t* xmm_val_addr = (int64_t*)&(uc->uc_mcontext.fpregs->_xmm[i]);
-    st->print_cr("XMM[%d]=" INTPTR_FORMAT " " INTPTR_FORMAT, i, xmm_val_addr[1], xmm_val_addr[0]);
+  size_t fpregs_offset = pointer_delta(uc->uc_mcontext.fpregs, uc, 1);
```

Comment:
Could the register substructure live outside uc on x64? If so, it may be safer to 
```suggestion
  size_t fpregs_offset = (uc->uc_mcontext.fpregs >= uc) ? pointer_delta(uc->uc_mcontext.fpregs, uc, 1) : 0;
```
or similar, since the register substructure may precede uc which would make pointer_delta assert.

Reply:
I think using 0 would require more changes to avoid accessing uc+0 which would be wrong. Richard and I already discussed about this above. We think it's acceptable. Some other projects claim that FP register substructure is inside the uc: https://github.com/mono/mono/blob/0f53e9e151d92944cacab3e24ac359410c606df6/mono/utils/mono-sigcontext.h#L263
We could also check the kernel code which writes it.

Reply:
After having read this, I have made the change. See commit nr. 7.